File: dbpro/src/main/java/kr/ac/sejong/feat/advertisement/AdDAO.java
package kr.ac.sejong.feat.advertisement;

import kr.ac.sejong.config.JDBCManager;

import java.util.List;

public class AdDAO {

	private AdDAO() {
	}

	private static AdDAO instance;

	public static AdDAO getInstance() {
		if (instance == null) {
			instance = new AdDAO();
		}
		return instance;
	}

	public List<AdDTO> selectAdvertisements() {
		final String query = "select AD_TITLE, " +
			"	AD_COMPANY, " +
			"	AD_DATE, " +
			"	AD_PRICE " +
			"from " +
			"	AD";

		return JDBCManager
			.getInstance()
			.queryForList(query, AdDTO.class);
	}

}

File: dbpro/src/main/java/kr/ac/sejong/feat/advertisement/AdDAO.kt
package kr.ac.sejong.feat.advertisement

import kr.ac.sejong.config.JDBCManager

class AdDAO private constructor() {
    companion object {
        @JvmStatic
        var instance: AdDAO? = null
            get() {
                if (field == null) {
                    field = AdDAO()
                }
                return field
            }
            private set
    }

    fun selectAdvertisements(): List<AdDTO> {
        val query = """
            SELECT 
                AD_TITLE,
                AD_COMPANY,
                AD_DATE,
                AD_PRICE
            FROM
            	AD
        """

        return JDBCManager
            .getInstance()
            .queryForList(query, AdDTO::class.java)
    }
}